### PR TITLE
Updated Crossdock tests not build Docker images for all architectures

### LINKED ISSUE #5350: [test] Crossdock tests do not need to build Docker images for all architectures
I just noticed in the [crossdock CI test logs](https://github.com/jaegertracing/jaeger/actions/runs/8649480331/job/23715804829?pr=5347):
```
#17 [linux/arm64 cert 2/2] RUN apk add --update --no-cache ca-certificates mailcap
#17 2.550 Executing ca-certificates-20240226-r0.trigger
#17 ...
#18 [linux/ppc64le stage-1 2/3] COPY --from=cert /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
#18 DONE 0.0s
#19 [linux/ppc64le stage-1 3/3] COPY --from=cert /etc/mime.types /etc/mime.types
#19 DONE 0.0s
#20 [linux/s390x cert 2/2] RUN apk add --update --no-cache ca-certificates mailcap
```
It means we're building images for arm64, ppc64le, s390x, etc. while we only need arm64 linux images for this test. We already have a shortcut for not building other architectures in the `docker-images` CI workflow for PRs, we should do the same for crossdock in all situations. https://github.com/jaegertracing/jaeger/blob/14aa43d4929ef344df722081aafd1a76c93c4f22/.github/workflows/ci-docker-build.yml#L53

### PR DESCRIPTION
**Pull request by varshith257**
## Which problem is this PR solving?

- Resolves #5350 

## Description of the changes
- Added the logic same in the docker-image CI to build linux/amd64 architecture only

## How was this change tested?
- Not yet

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [ ] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [ ] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

**Comment by varshith257** [comment]
@yurishkuro PTAL and suggest any needed to add.

File: .github/workflows/ci-crossdock.yml [hunk 1]
```diff
@@ -44,6 +44,10 @@ jobs:
       run: make install-ci
 
     - uses: docker/setup-qemu-action@68827325e0b33c7199eb31dd4e31fbe9023e06e3 # v3.0.0
+  
+    - name: Build only linux/amd64 crossdock image for Pull Request
+      if: github.ref_name != 'main'
```
Comment (yurishkuro) [inline .github/workflows/ci-crossdock.yml:49]: we don't need this check - crossdock *never* needs to be built for other architectures, since it's not an artifact that we release, it's a temporary set of containers just for testing

File: scripts/build-crossdock.sh [hunk 2]
```diff
@@ -5,8 +5,23 @@ set -euxf -o pipefail
 BRANCH=${BRANCH:?'missing BRANCH env var'}
 COMMIT=${GITHUB_SHA::8}
 
+mode=${1-main}
+
 make build-and-run-crossdock
 
+if [[ "$mode" == "pr-only" ]]; then
```
Comment (yurishkuro) [inline scripts/build-crossdock.sh:12]: you are adding more steps, not changing how existing steps behave. Wouldn't the previous command still build everything?

Comment (varshith257) [reply scripts/build-crossdock.sh:12]: Ok, noted. Will incorporate the requested changes

**Comment by yurishkuro** [comment]
this is the command that runs in the build (still lists all platforms):
```
docker buildx build -t localhost:5000/debugimg_alpine:latest --push \
	--build-arg golang_image=golang:1.22-alpine \
	--platform=linux/amd64,linux/s390x,linux/ppc64le,linux/arm64 \
```

how exactly are you testing your change?

**Comment by varshith257** [comment]
Testing those changes in different ways. Once it gets what we intended, I will push the related changes

**Comment by yurishkuro** [comment]
> Can guide where this docker cd is running?

you can see it in the checks in the PR. You can also check .github/workflows for what commands are being run.

File: scripts/build-crossdock.sh [hunk 3]
```diff
@@ -16,7 +16,7 @@ if [[ "$BRANCH" == "main" ]]; then
   bash scripts/docker-login.sh
   docker buildx build --push \
     --progress=plain \
-    --platform=linux/amd64 \
+    --platform=linux/arm64 \
```
Comment (yurishkuro) [inline scripts/build-crossdock.sh:19]: why is this changed?

File: docker/Makefile [hunk 4]
```diff
@@ -14,7 +14,7 @@ create-baseimg: prepare-docker-buildx
 	docker buildx build -t $(BASE_IMAGE) --push \
 		--build-arg root_image=$(ROOT_IMAGE) \
 		--build-arg cert_image=$(CERT_IMAGE) \
-		--platform=$(PLATFORMS) \
+		--platform=linux/arm64 \
```
Comment (yurishkuro) [inline docker/Makefile:17]: This will break multi-platform builds for other binaries

File: Makefile.Crossdock.mk [hunk 5]
```diff
@@ -32,7 +32,7 @@ build-crossdock-fresh: build-crossdock-linux
 	make crossdock-fresh
 
 .PHONY: crossdock-docker-images-jaeger-backend
-crossdock-docker-images-jaeger-backend: create-baseimg create-debugimg
+crossdock-docker-images-jaeger-backend: crossdock-create-baseimg crossdock-create-debugimg
```
Comment (yurishkuro) [inline Makefile.Crossdock.mk:35]: Have you tried just overriding the variable?
```suggestion
crossdock-docker-images-jaeger-backend: PLATFORMS=linux/amd64
crossdock-docker-images-jaeger-backend: create-baseimg create-debugimg
```

Comment (varshith257) [reply Makefile.Crossdock.mk]: @yurishkuro Yeah! It's working!  There is a lot to learn and explore under your guidance in this distributed platform :)

**Comment by varshith257** [comment]
Closing as added unnecesaary commits
